crates/nu-engine/src/documentation.rs

```diff
@@ -209,6 +209,34 @@ fn get_documentation(
         } else {
             let _ = write!(long_desc, "\n  > {}\n", example.example);
         }
+
+        match &example.result {
+            Some(result) => {
+                let decl_id = engine_state
+                    .find_decl("table".as_bytes(), &[])
+                    .expect("could not find `table` in the engine state");
+                let table = engine_state
+                    .get_decl(decl_id)
+                    .run(
+                        engine_state,
+                        stack,
+                        &Call::new(Span::new(0, 0)),
+                        PipelineData::Value(result.clone(), None),
+                    )
+                    .expect("could not run `table` on the output `Value` of the example");
```

Comment:
> "Nobody `.expect()`s the Spanish Inquisition"
	- Monty ~Python~ Rust

That could be a bit dangerous here. @stormasm is trying to separate out `nu-command` in smaller crates where some commands might not be available. And we were discussing that for example in `nana` the shell/terminal focussed `table` command might not be recommended.

I think instead of panicking through `.expect()` we can just ignore trying to perform an output if this feature is not available for now. For that you can use for example `Option::and_then()` or `Option::map()` or their `Result` equivalents if you don't want a big `if let` soup.
https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.and_then

Reply:
oooh wow, that a really good review comment :open_mouth: 

it was designed as a very temporary measure to have a good enough compiling code and i forgot to put it in the limitations above :eyes: 

i'll fix this not to have a dangling `expect` in the middle of nowhere :+1:

Reply:
@sholderbach 
so you would wrap the whole `let table = ...;` and `for ... { ... }` inside something like `.and_then()` or `.map()` called on `EngineState::find_decl` and `Command::run`?

Reply:
there would then be two nested `.and_then()` or `.map()`, right?

Reply:
If you don't mind I would push my approach on to your branch @amtoine. Was lazy to use rust-analyzer to help my backseat coding.

Reply:
i don't mind at all @sholderbach, please do push what you have in mind :relieved:

Reply:
Mhh push failed. Either my multi-remote git foo was not up to speed or it is disabled for your PR/repo.

Reply:
```diff
diff --git a/crates/nu-engine/src/documentation.rs b/crates/nu-engine/src/documentation.rs
index be10d595b..cec962e30 100644
--- a/crates/nu-engine/src/documentation.rs
+++ b/crates/nu-engine/src/documentation.rs
@@ -210,32 +210,30 @@ fn get_documentation(
             let _ = write!(long_desc, "\n  > {}\n", example.example);
         }

-        match &example.result {
-            Some(result) => {
-                let decl_id = engine_state
-                    .find_decl("table".as_bytes(), &[])
-                    .expect("could not find `table` in the engine state");
-                let table = engine_state
-                    .get_decl(decl_id)
-                    .run(
-                        engine_state,
-                        stack,
-                        &Call::new(Span::new(0, 0)),
-                        PipelineData::Value(result.clone(), None),
-                    )
-                    .expect("could not run `table` on the output `Value` of the example");
+        if let Some(result) = &example.result {
+            let table = engine_state
+                .find_decl("table".as_bytes(), &[])
+                .and_then(|decl_id| {
+                    engine_state
+                        .get_decl(decl_id)
+                        .run(
+                            engine_state,
+                            stack,
+                            &Call::new(Span::new(0, 0)),
+                            PipelineData::Value(result.clone(), None),
+                        )
+                        .ok()
+                });

-                for item in table {
-                    let _ = writeln!(
-                        long_desc,
-                        "  {}",
-                        item.into_string("", engine_state.get_config())
-                            .replace('\n', "\n  ")
-                            .trim()
-                    );
-                }
+            for item in table.into_iter().flatten() {
+                let _ = writeln!(
+                    long_desc,
+                    "  {}",
+                    item.into_string("", engine_state.get_config())
+                        .replace('\n', "\n  ")
+                        .trim()
+                );
             }
-            None => {}
         }
     }
```

Reply:
> Mhh push failed. Either my multi-remote git foo was not up to speed or it is disabled for your PR/repo.
>
oh yeah, the branches are not protected, but that's a personal repo, so unless i change the settings, i'm the only one allowed to push on `amtoine/nushell` :thinking:

Reply:
let me check the patch :smirk:

Reply:
Let me know if you want it in `git apply`/`git am` ready format for Linux kernel dev level convenience

Reply:
> Let me know if you want it in `git apply`/`git am` ready format for Linux kernel dev level convenience
>
what do you mean?

i created a file called `diff.patch` and used the GNU `patch` as
```bash
open diff.patch | patch -p1
```
:thinking:

Reply:
love the patch you came up with!
i was close but not sure at all i was in the right direction :eyes: 

the only question i have is about the `.ok()` on the result of `Command::run`, we're just skipping a potential `ShellError` here, right?

Reply:
patch applied in fc415bb :ok_hand: 
`cargo test --workspace` is currently running on my machine to make sure everything is alright.

Reply:
`.ok()` turns a `Result::Err` into an `Option::None` and that gets flattened with the `.and_then()` so yes we ignore any errors from the `table` execution.

The part I am not too proud of is the `.into_iter().flatten()` but hey works and keeps the diff smaller.

Reply:
oooh ok i see, that's fine then :ok_hand: